treeherder/perf/management/commands/import_perf_data.py

```diff
@@ -56,22 +57,43 @@ def _add_series(pc, project_name, signature_hash, signature_props, verbosity):
             'option_collection': option_collection,
             'platform': platform,
             'framework': framework,
-            'extra_properties': extra_properties
+            'extra_properties': extra_properties,
+            'last_updated': datetime.datetime.fromtimestamp(0)
         })
 
     series = pc.get_performance_data(
         project_name, signatures=signature_hash,
         time_interval=PerformanceTimeInterval.ONE_YEAR)[signature_hash]
 
-    with transaction.atomic():
+    try:
+        new_series = []
+        latest_timestamp = datetime.datetime.fromtimestamp(0)
         for datum in series:
-            PerformanceDatum.objects.get_or_create(
+            timestamp = datetime.datetime.fromtimestamp(datum['push_timestamp'])
+            new_series.append(PerformanceDatum(
                 repository=repository,
                 result_set_id=datum['result_set_id'],
                 job_id=datum['job_id'],
                 signature=signature,
                 value=datum['value'],
-                push_timestamp=datetime.datetime.fromtimestamp(datum['push_timestamp']))
+                push_timestamp=timestamp))
+            if timestamp > latest_timestamp:
+                latest_timestamp = timestamp
+
+        PerformanceDatum.objects.bulk_create(new_series)
+        signature.last_updated = latest_timestamp
+        signature.save()
+
+    except IntegrityError:
+        with transaction.atomic():
```

Comment:
Could you add a comment here explaining that the bulk_update process won't work if the data we downloaded overlaps with existing data, and we need to fall back to creating items one-by-one?